addons/l10n_it_edi/tests/test_res_company.py

```diff
@@ -0,0 +1,18 @@
+from odoo.tests import tagged
+from odoo.addons.l10n_it_edi.tests.common import TestItEdi
+from odoo.tests.common import Form
+
+@tagged('post_install_l10n', 'post_install', '-at_install')
+class TestResCompany(TestItEdi):
+
+    def test_tax_representative(self):
+        """ Test to check for l10n_it_edi_tax_representive if l10n_it_has_tax_representative is not set."""
+
+        self.company.write({
+            "l10n_it_tax_representative_partner_id": self.italian_partner_a.id,
+            "l10n_it_has_tax_representative": True,
+            })
+        with Form(self.company) as company:
+            company.l10n_it_has_tax_representative = False
+
+        self.assertFalse(self.company.l10n_it_tax_representative_partner_id)
```

Comment:
Please test your issue is fixed instead of testing the fix implemented works

> When a xml file is generated from the invoice, the company tax code is used and not the previous tax representative, after the setting has changed.

So generate the xml and test it's properly set if you remove the `l10n_it_has_tax_representative` the xml generated is working as inteded